File: Prog/Count_prefix/Count_Prefix.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int countPrefixes(char** words, int wordsSize, char* s) {
    int count = 0;
    
    for (int i = 0; i < wordsSize; i++) {
        if (strncmp(s, words[i], strlen(words[i])) == 0) {
            count++;
        }
    }
    
    return count;
}

int main() {
    char* words[] = {"a", "b", "c", "ab", "bc", "abc"};
    int wordsSize = sizeof(words) / sizeof(words[0]);
    char* s = "abc";
    int result = countPrefixes(words, wordsSize, s);
    printf("%d\n", result);
    
    return 0;
}

File: Prog/Count_prefix/Count_Prefix.py
def countPrefixes(words, s):
    count = 0
    
    for word in words:
        if s.startswith(word):
            count += 1
    
    return count

# Example usage
words = ["a", "b", "c", "ab", "bc", "abc"]
s = "abc"
result = countPrefixes(words, s)
print(result)
